### PR TITLE
Added data.sh to SPRING-INX IITM Recipe 

### PR DESCRIPTION
**Pull request by arjun-gangwar**
## What?

Added egs2/spring_speech/local/data.sh for data preparation in stage 1.

## Why?

The .gitignore in espnet repository had egs*/\*/\*/data* which prevented data.sh from being committed to the repository.

## See also

Please refer PR #5505

### CODE REVIEW

File: egs2/spring_speech/local/data.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+SECONDS=0
+
+lang=
+release_name="SPRING_INX"
```
Comment (ftshijt) [inline egs2/spring_speech/local/data.sh:15]: Why not directly use SPRING_INX in the following context?

Comment (arjun-gangwar) [reply egs2/spring_speech/local/data.sh:15]: The name of the file being downloaded will change depending on the lang argument in run.sh and the versions maintained in our back-end. We are using release_name for relative names of the files.

**Comment by ftshijt** [comment]
Hi, thanks for your update. Could you please update your branch with the latest master branch? We just fixed a CI issue, which has been also causing blocking in this PR.

**Comment by arjun-gangwar** [comment]
Hi, I have updated the branch. Thanks.

File: egs2/spring_speech/local/data.sh [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+data_url="https://asr.iitm.ac.in/SPRING_INX/data/${release_name}_${lang}"
+
+if [ $# -ne 0 ]; then
+    log "Error: No positional arguments are required."
+    exit 2
+fi
+
+if [ -z "${release_name}" ]; then
+    log "Fill the value of '${release_name}' of db.sh"
```
Comment (ftshijt) [inline egs2/spring_speech/local/data.sh:38]: I feel this log msg is incorrect. Could you double check?

**Comment by sw005320** [comment]
Hi @arjun-gangwar,

Again, can you follow the other recipe’s README.md and add the pre-trained model links (HF hub), configs, and environmental information?
See https://github.com/espnet/espnet/blob/master/egs2/librispeech_100/asr1/README.md
Usually configs and environmental information are generated from README.md in the experimental directory.

**Comment by arjun-gangwar** [comment]
Hi @sw005320,
We are working on collating the required data. I will create another PR for adding proper README and HF model links as discussed in PR #5505. This PR was just to add data.sh file in recipe without which the existing recipe would not work. Thanks.

File: egs2/spring_speech/local/data.sh (hunk 2 above)
Comment (arjun-gangwar) [reply egs2/spring_speech/local/data.sh:38]: Yes, I will correct it. Thanks.

**Pull request merged**